Brief-over-of-Heads-and-firmware-security/FAQ.md

```diff
@@ -0,0 +1,139 @@
+====
+
+Heads is an open source firmware, OS configuration and hardware hardening guide
+ for building slightly more secure systems. Installing Heads requires opening
+ the machine and extensive warranty voiding, so this document tries to answer
+ some of the questions about it.
+
+Why replace UEFI with coreboot
+----
+
+While Intel's edk2 tree that is the base of UEFI firmware is open source, the
+ firmware that vendors install on their machines is proprietary and closed
+ source. Updates for bugs fixes or security vulnerabilities are at the vendor's
+ convenience; user specific enhancements are likely not possible; and the code
+ is not auditable.
+
+UEFI is much more complex than the BIOS that it replaced. It consists of
+ millions of lines of code and is an entire operating system, with network
+ device drivers, graphics, USB, TCP, https, etc, etc, etc. All of these features
+ represents increased "surface area" for attacks, as well as unnecessary
+ complexity in the boot process.
+
+coreboot is open source and focuses on just the code necessary to bring the
+ system up from reset. This minimal code base has a much smaller surface area
+ and is possible to audit. Additionally, self-help is possible if custom
+ features are required or if a security vulnerability needs to be patched.
+
+What's wrong with UEFI Secure Boot
+----
+
+Can't audit it, signing keys are controlled by vendors, doesn't handle hand off
+ in all cases, depends on possible leaked keys.
+
+Why use Linux instead of vboot2
+----
+
+vboot2 is part of the coreboot tree and is used by Google in the Chromebook
+ system to provide boot time security by verifying the hashes on the coreboot
+ payload. This works well for the specialized Chrome OS on the Chromebook, but
+ is not as flexible as a measured boot solution.
+By moving the verification into the boot scripts we're able to have a much
+ flexible verification system and use more common tools like PGP to sign
+ firmware stages.
+
+What about Trusted GRUB
+----
+
+The mainline grub doesn't have support for TPM and signed kernels, but there is
+ a Trusted grub fork that does. Due to philosophical differences the code might
+ not be merged into the mainline. And due to problems with secure boot (which
+ Trusted Grub builds on), many distributions have signed insecure kernels
+ that bypass all of the protections secure boot promised.
+Additionally, grub is closer to UEFI in that it must have device drivers for
+ all the different boot devices, as well as filesystems. This duplicates the
+ code that exists in the Linux kernel and has its own attack surface.
+
+Using coreboot and Linux as a boot loader allows us to restrict the signature
+ validation to keys that we control. We also have one code base for the device
+ drivers in the Linux-as-a-boot-loader as well as Linux in the operating system.
+
+What is the concern with the Intel Management Engine
+----
+
+"Rootkit in your chipset", "x86 considered harmful", etc
```

Comment:
[Rootkit in your laptop](me.bios.io/images/c/ca/Rootkit_in_your_laptop.pdf):
Rootkit in your laptop: Hidden code in your chipset and how to discover what exactly it does Igor Skochinsky Hex-Rays Breakpoint 2012 Melbourne

[Intel x86 considered harmful](https://blog.invisiblethings.org/papers/2015/x86_harmful.pdf) Oct 27, 2015 by Joanna Rutkowska

Reply:
I thought these seemed weirdly vague.  The links do not appear on https://trmm.net/Heads_FAQ nor do they appear in previous versions of the page.  Same for https://github.com/osresearch/heads/blob/master/FAQ.md, Where did you reference the links from?

Reply:
https://github.com/osresearch/heads-wiki/pull/49/files/1899083cd24a471d83b67df360b4a174165d8bd3#diff-a9fcdfdf121c9e5fe20d9a12db884bd5R71
http://2012.ruxconbreakpoint.com/assets/Uploads/bpx/Breakpoint%202012%20Skochinsky.pdf
https://recon.cx/2014/slides/Recon%202014%20Skochinsky.pdf